python/tvm/driver/tvmc/compiler.py

```diff
@@ -356,6 +363,13 @@ def compile_model(
         if codegen["config_key"] is not None:
             config[codegen["config_key"]] = codegen_from_cli["opts"]
 
+    timing_inst = None
```

Comment:
`compile_model` is also used by the TVMC Python API, so if a user adds `PassTimingInstrument` to their list of instruments and calls `compile_model` providing the list of instruments, they may get an unexpected output printed to the console. 

Since it is relatively easy for a python API user to add the instrument and print it themselves, I think it would be better to move the printing functionality up to `drive_compile` so that the feature is only exposed to the command line API. It would also simplify checking if the option was enabled.

Happy to hear more thoughts from others though.

Reply:
We discussed this offline. Because the `PassTimingInstrument` object clears its timing profile (retrieved using `.render()`) upon exiting the `PassContext` used in `compile_model()`, thus making the results unavailable outside of its scope, the decision was to leave the printing functionality inside `compile_model()`.  
However, to facilitate using this functionality through the TVMC Python API and avoid any unexpected outputs, a `print_pass_times` bool parameter would be added to `compile_model()`, meaning the python API user does not need to add the timing instrument themselves.